x-pack/plugins/apm/public/components/app/tutorials/agent_config_instructions.tsx

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { EuiCodeBlock, EuiSpacer } from '@elastic/eui';
+import {
+  getApmAgentCommands,
+  getApmAgentVariables,
+  getApmAgentLineNumbers,
+  getApmAgentHighlightLang,
+} from './commands/get_apm_agent_commands';
+import { AgentConfigurationTable } from './agent_config_table';
+
+export function AgentConfigInstructions({
+  variantId,
+  apmServerUrl,
+  secretToken,
+  apiKey,
+}: {
+  variantId: string;
+  apmServerUrl?: string;
+  secretToken?: string;
+  apiKey?: string;
+}) {
+  const defaultValues = {
+    apmServiceName: 'my-service-name',
+    apmEnvironment: 'my-environment',
+  };
```

Comment:
Why does this needs to be variables?

Reply:
We can, but this way, instead of adding them in every Agent Instructions, i was following the DRY principle here which means, when we do `Moustache.render` for all the templates, all of them get this default value.

Since we have the same default value for all agents, this made sense. If different agents propose different default values, then it would make sense to directly port them inside Instructions directly

Reply:
I'm not sure DRY is better in this case. I don't think we strictly need the same defaults. Again, the most important is the ease of updating individual agent instructions where there are as few side effects to other instructions as possible.